core/src/main/java/org/infinispan/stream/impl/local/AbstractLocalCacheStream.java

```diff
@@ -294,9 +295,16 @@ public boolean noneMatch(Predicate<? super R> predicate) {
 
    @Override
    public CloseableIterator<R> iterator() {
-      return Closeables.iterator(getOrCreateStream());
+      // If the stream is null we can assume no intermediate operations and thus we can support remove
+      if (stream == null) {
+         return removableIterator(Closeables.iterator(getOrCreateStream()));
```

Comment:
nitpick: if the stream is null, isn't worth it to invoke a ```createStream``` to avoid double checking? 
the same logic could be applied to the else branch by passing the stream directly.